index.js

```diff
@@ -209,6 +211,17 @@ module.exports = class Hyperdrive extends ReadyResource {
       : Promise.resolve(name)
   }
 
+  watch (folder) {
+    folder = normalizePath(folder || '/')
+
+    if (folder.endsWith('/')) folder = folder.slice(0, -1)
+
+    const range = { gt: folder + '/', lt: folder + '0' }
+    const options = { map: this.checkout }
```

Comment:
did you mean to bind this function?

Reply:
does the watcher close the checkouts also or are they leaking atm?

Reply:
Yes, I meant to bind it, before I was not doing it but it's the same effect

It's same behaviour even if you pass a `map`, is that what you're saying?

Reply:
it def does not work with a non-bound func. dunno what you mean with second comment, but who closes the drive checkout when the snapshot is thrown away?

Reply:
Like it should not leak because it's closing the sub now:
https://github.com/holepunchto/hyperdrive-next/blob/f518866874450a51f563eadf5b91a2865be0370b/index.js#L97

I think closing the main bee instance doesn't close the sub watchers, right?

Reply:
Does the watcher close the checkout?

Reply:
One sec, my previous comment was at the same time so I didn't answer your question

Reply:
Let me first fix the new error `Can only watch the main bee instance`